src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,82 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    private SessionFactory sessionFactory;
+
+    public MovieSessionDaoImpl() {
+        sessionFactory = HibernateUtil.getSessionFactory();
+    }
+
+    @Override
+    public MovieSession add(MovieSession entity) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(entity);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot save MovieSession entity "
+            + entity + " in DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return entity;
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long movieSessionId) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<MovieSession> getMovieSessionQuery = session.createQuery(

```

Comment:
It’s good practice if you have similar naming for similar methods in different classes. better to give noun names for query